src/protocols/ssh/client.c

```diff
@@ -70,8 +70,12 @@ int guac_ssh_client_free_handler(guac_client* client) {
 
     /* Free terminal (which may still be using term_channel) */
     if (ssh_client->term != NULL) {
-        guac_terminal_free(ssh_client->term);
+        /* Stop the terminal to terminate ssh_input_thread */
```

Comment:
This is somewhat confusing, as the `pthread_join()` call which immediately follows deals with the client thread. Perhaps there is a way to word this such that it is more relevant?